lib/streamlit/elements/exception.py

```diff
@@ -63,32 +63,45 @@ def exception(self, exception: BaseException) -> DeltaGenerator:
         >>> st.exception(e)
 
         """
-        exception_proto = ExceptionProto()
-        marshall(exception_proto, exception)
-        return self.dg._enqueue("exception", exception_proto)
+        return send_proto(self.dg, exception, is_uncaught_app_exception=False)
 
     @property
     def dg(self) -> DeltaGenerator:
         """Get our DeltaGenerator."""
         return cast("DeltaGenerator", self)
 
 
-def marshall(exception_proto: ExceptionProto, exception: BaseException) -> None:
+def send_proto(
```

Comment:
nit: I guess I would name this `_send_proto` to indicate that we only use it internally and don't intend others to use it

Reply:
OK, I was thinking of `_` prefix as more internal to this class/file, but I'll update this.